src/__tests__/styled.test.js

```diff
@@ -146,4 +146,14 @@ describe('styled', () => {
 
         expect(vnode).toMatchVNode('tag', { className: 'go', bar: 1 });
     });
+
+    it('pass truthy logical and operator', () => {
```

Comment:
Actually can be dropped since it's kind of verbose. I just added this here to verify the `&&` operator

Reply:
But this test does not run with TypeScript 🤔  does it surface the same problem in JavaScript as well?

Reply:
No, it's working like expected. In general this "issue" can be solved by just ignoring the tsc on the user's end (aka adding `@ts-ignore`)

Reply:
Oh, I see! 😄  Do you think we can capture it in here instead? https://github.com/cristianbote/goober/blob/master/ts-tests/test-api.tsx

Reply:
Okay I added them there. Undoing the `global.d.ts` update of this PR will make the TSC "complain" for the `<BoxWithLogicalAnd />`